### ISSUE TITLE
Dokuwiki searching the systems FULL directory structure

### ISSUE THREAD

**Issue by Pontiac76** [splitbrain/dokuwiki#1452]
I haven't quite figured out exactly what extension it is yet, but, I've noticed that the search function in inc/search.php doesn't take into consideration that if $base and $dir are empty, the entire directory structure of the file system (from root directory) can be searched.

I'm modifying my version of the code to look at $base and if empty, substitute the $base as the main directory to dokuwiki.

I ran across this because I was running into time out issues when going to a particular page.  I was thinking I had a bad symlink or something that was causing circular lookups and getting myself into an infinite loop.  I added this to the first line of the search function:

file_put_contents('/tmp/dokuwiki_search.txt',"$lvl -> Base=$base -> Dir=$dir\n",FILE_APPEND);

The first few lines:
1 -> Base= -> Dir=
2 -> Base= -> Dir=/bin
2 -> Base= -> Dir=/boot
3 -> Base= -> Dir=/boot/grub
4 -> Base= -> Dir=/boot/grub/locale
2 -> Base= -> Dir=/build

... uhhh... no.... Not supposed to be there.... ;)

I think it is a plugin that is not passing the right information to this function.  However, as a safeguard, I'd suggest that if $dir doesn't contain at least the root directory of the Dokuwiki installation path, it should just abort and not continue, unless an optional parameter is added that'll override.

**Comment by splitbrain** [comment]
I think a safe guard makes sense. Pull Request welcome. It would also be nice to know what part or plugin caused the trouble.

**Comment by Pontiac76** [comment]
Submitted a forced path, no override, but easy to add.

The plugin that triggered this was ORPHANS WANTED.  On their github page (https://github.com/lupo49/dokuwiki-plugin-orphanswanted/blob/master/helper.php#L174) it does a search with $conf.  When I tried using my file_put_contents and output print_r($conf,true) to /tmp/somefile, the result file was empty.  I then tried $conf['datadir'] as indicated and again empty.  I then did a print_r(get_defined_vars(), true) and conf didn't show up anywhere.  So for some reason my instance of dokuwiki doesn't have $conf and it should according to https://www.dokuwiki.org/devel:environment#conf but everything else is working.
